src/algebra/module/localized_module.lean

```diff
@@ -387,4 +387,108 @@ instance localized_module_is_localized_module :
+{ to_fun := λ p, p.lift_on (λ x, (is_localized_module.map_units f x.2).unit⁻¹ (f x.1))
+  begin
+    rintros ⟨a, b⟩ ⟨a', b'⟩ ⟨c, eq1⟩,
+    dsimp,
+    generalize_proofs h1 h2,
+    erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, ←h2.unit⁻¹.1.map_smul,
+      module.End_algebra_map_is_unit_inv_apply_eq_iff', ←linear_map.map_smul, ←linear_map.map_smul],
+    exact ((is_localized_module.eq_iff_exists S f).mpr ⟨c, eq1⟩).symm,
+  end,
+  map_add' := λ x y,
+  begin
+    induction x using localized_module.induction_on with a b,
+    induction y using localized_module.induction_on with a' b',
+    simp only [localized_module.mk_add_mk, localized_module.lift_on_mk],
+    generalize_proofs h1 h2 h3,
+    erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, smul_add, ←h2.unit⁻¹.1.map_smul,
+      ←h3.unit⁻¹.1.map_smul, map_add],
+    congr' 1,
+    all_goals { erw [module.End_algebra_map_is_unit_inv_apply_eq_iff'] },
+    { dsimp, erw [mul_smul, f.map_smul], refl, },
+    { dsimp, erw [mul_comm, f.map_smul, mul_smul], refl, },
+  end,
+  map_smul' := λ r x,
+  begin
+    induction x using localized_module.induction_on with a b,
+    erw [localized_module.lift_on_mk, localized_module.lift_on_mk, one_mul],
+    generalize_proofs h1,
+    dsimp,
+    erw [f.map_smul, h1.unit⁻¹.1.map_smul],
+    congr,
+  end }
+
+lemma from_localized_module.inj : function.injective $ from_localized_module S f :=
+λ x y eq1,
+begin
+  induction x using localized_module.induction_on with a b,
+  induction y using localized_module.induction_on with a' b',
+  simp only [from_localized_module, linear_map.coe_mk, localized_module.lift_on_mk] at eq1,
+  generalize_proofs h1 h2 at eq1,
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, ←linear_map.map_smul,
+    module.End_algebra_map_is_unit_inv_apply_eq_iff'] at eq1,
+  erw [localized_module.mk_eq, ←is_localized_module.eq_iff_exists S f, f.map_smul, f.map_smul, eq1],
+  refl,
+end
+
+lemma from_localized_module.surj : function.surjective $ from_localized_module S f :=
+λ x, let ⟨⟨m, s⟩, eq1⟩ := is_localized_module.surj S f x in ⟨localized_module.mk m s,
+by { rw [from_localized_module, linear_map.coe_mk, localized_module.lift_on_mk,
+    module.End_algebra_map_is_unit_inv_apply_eq_iff, ←eq1], refl }⟩
+
+lemma from_localized_module.bij : function.bijective $ from_localized_module S f :=
+⟨from_localized_module.inj _ _, from_localized_module.surj _ _⟩
+
+/--
+If `(M', f : M ⟶ M')` satisfies universal property of localized module, then `M'` is isomorphic to
+`localized_module S M` as an `R`-module.
+-/
+@[simps] noncomputable def iso : localized_module S M ≃ₗ[R] M' :=
+{ ..from_localized_module S f,
+  ..equiv.of_bijective (from_localized_module S f) $ from_localized_module.bij _ _}
+
+lemma iso_apply_mk (m : M) (s : S) :
+  iso S f (localized_module.mk m s) = (is_localized_module.map_units f s).unit⁻¹ (f m) :=
+rfl
+
+private lemma iso_symm_apply_aux (m : M') :
```

Comment:
Can you remove the `private` attribute? Even if the lemma will never be used by anyone else we can keep it visible.

Reply:
done in [7be68f9](https://github.com/leanprover-community/mathlib/pull/15559/commits/7be68f9febb81fb9337a57cb7e290992db6053dc)